vllm/config.py

```diff
@@ -316,6 +317,15 @@ def __init__(
         self._verify_cuda_graph()
         self._verify_bnb_config()
 
+    def compute_hash(self):
+        # summarize all the factors that affect the model compulation and
+        # hence model compilation, and use them to generate a hash
```

Comment:
These look like they could plausibly affect the compute graph.

`revision` will affect the model that's loaded from HF, and rope scaling and theta could be baked in as constants. `quantization_param_path` is used for kv cache quantization
```
revision
code_revision
rope_scaling
rope_theta
max_model_len
quantization_param_path
max_logprobs
disable_sliding_window
```

Reply:
why would `max_logprobs` affect the computation?

I don't think `disable_sliding_window` affect the computation.

for rope-related field, I agree we should add it to the factors.

we hash the fx graph, as long as the config does not change the fx graph's structure, then it is safe to reuse the compilation.

therefore, anything inside the attention layer does not matter. the attention op is wrapped as a custom op.